src/ZkFold/Cardano/UPLC/Rollup.hs

```diff
@@ -98,22 +101,27 @@ rollup (RollupSetup ledgerRules dataCurrency threadValue feeAddress) (UpdateRoll
     && sort update' == sort update
 
     -- Check the current rollup output
-    && val == threadValue
+    -- && val == threadValue  -- Doesn't work because 'val' also contains lovelaces
+    && member threadCurrency (getValue val)
 
     -- Check the next rollup output
-    && out' == TxOut addr threadValue (OutputDatum (Datum $ mkI state')) Nothing
+    -- && out' == TxOut addr threadValue (OutputDatum (Datum $ mkI state')) Nothing
+    && member threadCurrency (getValue $ txOutValue out')
+    && txOutAddress out' == addr
+    && txOutDatum out'   == (OutputDatum . Datum . mkI $ state')
 
     -- Check the fee output
     && case outFee of
       TxOut addr'' _ NoOutputDatum Nothing -> feeAddress == addr''
       _                                    -> False
-rollup (RollupSetup _ _ threadValue _) ForwardValidation ctx =
-  let
-    out = head $ txInfoOutputs $ scriptContextTxInfo ctx
-  in
-    txOutValue out == threadValue
--- TODO: implement other cases
-rollup _ _ _ = False
+
+-- rollup (RollupSetup _ _ threadValue _) ForwardValidation ctx =
```

Comment:
Disabled other choices of redeemer.  In any case, having separate definitions of rollup (pattern matching different choices for the redeemer) does not seem to work, as I get an error about the script size.  Hopefully using `case rollupRedeemer of` inside a single definition of `rollup` solves this issue.